Fix GraphQL error when duplicating M2A items with same name fields of conflicting types. (#26367)
directus/directus: .changeset/fix-m2a-duplication-graphql-error.md

---
"@directus/app": patch
---

> Fixed GraphQL validation error when duplicating items with M2A fields that link to collections with conflicting field types by using field aliases in M2A inline fragments and transforming the response back to original field names.

Reviewer: Ideally we reduce this to just the fixed issue, e.g.:
 ```suggestion
Fixed GraphQL error when duplicating M2A items with same name fields of conflicting types.
```

---
"@directus/app": patch
---

Fixed GraphQL validation error when duplicating items with M2A fields that link to collections with conflicting field types.

When duplicating items that have M2A (Many-to-Any) relations pointing to different collections where those collections have fields with the same name but different types (e.g., one JSON field and one relation field both named "items"), GraphQL would throw a validation error because inline fragments require compatible types for fields with the same name.

> This fix uses field aliases (prefixed with the collection name) in the generated GraphQL query for M2A inline fragments, then transforms the response data back to use the original field names before creating the duplicated item.

Reviewer 2: Hey @dstockton :tada: Thank you for your PR! Let’s use only the first sentence for the changeset, to keep the release notes clear and concise. Additional lines should be reserved for breaking changes or important warnings.

```suggestion
Fixed GraphQL validation error when duplicating items with M2A fields that link to collections with conflicting field types.
```
